### PR TITLE
Trigger installation of the Managed Cloud SDK on selection

### LINKED ISSUE #2808: Trigger installation when applying changes to the Google Cloud SDK Preference page
Trigger installation when unchecking _Use local SDK_.
screen shot 2018-02-14 at 3 09 47 pm

### PR DESCRIPTION
**Pull request by briandealwis**
Required making `CloudSdkManager` mockable.

Fixes #2808

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkManager.java [hunk 1]
```diff
@@ -41,10 +41,21 @@
       "com.google.cloud.tools.eclipse.sdk/enable.managed.cloud.sdk";
 
   // To be able to write tests for the managed Cloud SDK feature, which is disabled at the moment.
-  @VisibleForTesting
-  public static boolean forceManagedSdkFeature;
+  @VisibleForTesting public static boolean forceManagedSdkFeature;
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkManager.java]: The amount of API purely for testing is making me a little uncomfortable.

File: plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkManager.java [hunk 2]
```diff
@@ -41,10 +41,21 @@
...
+
+  @VisibleForTesting public static CloudSdkManager singleton;
+
+  public static synchronized CloudSdkManager getDefault() {
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkManager.java]: getInstance is customary name for this.

File: plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkManager.java [hunk 3]
```diff
@@ -41,10 +41,21 @@
...
+    if (singleton == null) {
+      singleton = new CloudSdkManager();
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkManager.java]: Can we make the constructor explicit and not public? As written this is not a true singleton. Other instances can be created.

File: plugins/com.google.cloud.tools.eclipse.sdk.test/src/com/google/cloud/tools/eclipse/sdk/CloudSdkManagerTest.java [hunk 4]
```diff
@@ -44,7 +44,7 @@
 public class CloudSdkManagerTest {
 
   @Mock private ManagedCloudSdk managedCloudSdk;
-  @Mock private IProgressMonitor monitor;
+  private CloudSdkManager fixture = new CloudSdkManager();
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.sdk.test/src/com/google/cloud/tools/eclipse/sdk/CloudSdkManagerTest.java]: Could be `final`?

File: plugins/com.google.cloud.tools.eclipse.sdk.ui/src/com/google/cloud/tools/eclipse/sdk/ui/preferences/CloudSdkPreferenceArea.java [hunk 5]
```diff
@@ -60,6 +60,7 @@
       "com.google.cloud.tools.eclipse.preferences.main"; //$NON-NLS-1$
 
   private Button chooseSdk;
+  private Button updateSdk;
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.sdk.ui/src/com/google/cloud/tools/eclipse/sdk/ui/preferences/CloudSdkPreferenceArea.java]: Unused for this PR.

**Pull request merged**